patches/minecraft/net/minecraft/client/renderer/ItemRenderer.java.patch

```diff
@@ -81,3 +100,23 @@
          }
  
          if (this.field_187469_f < 0.1F)
+@@ -617,4 +629,19 @@
+             this.field_187471_h = 0.0F;
+         }
+     }
++
++    private boolean forgeFluidOverlay(float partialTicks)
++    {
++        Block viewBlock = ActiveRenderInfo.func_186703_a(this.field_78455_a.field_71441_e, this.field_78455_a.func_175606_aa(), partialTicks).func_177230_c();
++        if (viewBlock instanceof net.minecraftforge.fluids.IFluidBlock)
++        {
++            ResourceLocation overlay = ((net.minecraftforge.fluids.IFluidBlock)viewBlock).getFluid().getOverlay();
++            if (overlay != null && !net.minecraftforge.event.ForgeEventFactory.renderWaterOverlay(field_78455_a.field_71439_g, partialTicks))
```

Comment:
That change would be unrelated to this PR.

But yes, getting the correct block to render the overlay for (instead of currently the entity center pos) would be nice.